c/tskit/trees.h

```diff
@@ -891,6 +891,39 @@ int tsk_treeseq_simplify(const tsk_treeseq_t *self, const tsk_id_t *samples,
     tsk_size_t num_samples, tsk_flags_t options, tsk_treeseq_t *output,
     tsk_id_t *node_map);
 
+/**
+@brief Extends edges
+
+Returns a modified tree sequence in which the span covered by ancestral nodes
+is "extended" to regions of the genome according to the following rule:
+If an ancestral segment corresponding to node `n` has parent `p` and
+child `c` on some portion of the genome, and on an adjacent segment of
+genome `p` is the immediate parent of `c`, then `n` is inserted into the
+edge from `p` to `c`. This involves extending the span of the edges
+from `p` to `n` and `n` to `c` and reducing the span of the edge from
+`p` to `c`. Since the latter edge may be removed entirely, this process
+reduces (or at least does not increase) the number of edges in the tree
+sequence.
+
+The method works by iterating over the genome to look for edges that can
```

Comment:
Debatable whether we want to bother documenting the C API for this also?

Reply:
Would save a bit of trouble worrying about , .e.g. whether ``max_iter`` should be an int, or some other type.

Reply:
So, we're adding this to the next C API release then? We should update the C changelog so.